### ISSUE TITLE
cluster.wait() & cluster.close() should have timeout parameter

### ISSUE THREAD

**Issue by wsfax** [pgiri/dispy#64]
cluster.wait() and cluster.close() both wait for the threading.Event called self._complete.

normally threading.Event takes a timeout parameter (see [here](https://docs.python.org/2/library/threading.html#threading.Event.wait))

Could you add this timeout parameter to both cluster.wait and cluster.close and forward it to self._complete.wait(timeout) ? 

This is very helpful in case the nodes do not respond anymore and you want to close the cluster and handle the timeout in a certain way

**Comment by wsfax** [comment]
here is a simple proposal:


```
    def wait(self, timeout=None):
        """Wait for scheduled jobs to complete.
        """
        return self._complete.wait(timeout)

    def close(self, timeout=None):
        """Close the cluster (jobs can no longer be submitted to
        it). If there are any jobs pending, this method waits until
        they all finish. Additional clusters may be created after this
        call returns.
        """
        if self._compute:
            ret = self.wait(timeout)
            Coro(self._cluster.del_cluster, self).value()
            self._compute = None
            return True if ret else False
```

**Comment by pgiri** [comment]
This is addressed in current github and in 4.7.2 release.

**Comment by pgiri** [comment]
Closing this as fixed.
